### PR TITLE
Add test cases to support systems with disaggregated resources

### PR DESCRIPTION
**Pull request by dongahn**
- Add a very large "theoretical" system built with the concept of disaggregated resources
- Add a few jobspecs that can match with these disaggregated resources
- Extend advanced test cases to make use of them (guard them with `LONGTEST` since the system configuration is as large as today's pre-exascale system like Sierra.)

Resolve Issue #459.

This should be merged after PR #455 is merged first.

### CODE REVIEW

File: t/t3006-resource-advanced.t [hunk 1]
```diff
@@ -37,4 +38,30 @@ test_expect_success "${test002_desc}" '
     test_cmp 002.R.out ${exp_dir}/002.R.out
 '
 
+#
+# Selection Policy -- High ID first (-P high)
+#     The resource vertex with higher ID is preferred among its kind
+#     (e.g., node0 is preferred over node1 if available)
```
Comment (SteVwonder) [inline t/t3006-resource-advanced.t]:
```suggestion
#     (e.g., node1 is preferred over node0 if available)
```

File: t/t3006-resource-advanced.t [hunk 2]
```diff
@@ -37,4 +38,30 @@ test_expect_success "${test002_desc}" '
...
+#
+cmds003="${cmd_dir}/cmds02.in"
+test003_desc="match for a large system with disaggregated resources (pol=high)"
+test_expect_success LONGTEST "${test003_desc}" '
+    sed "s~@TEST_SRCDIR@~${SHARNESS_TEST_SRCDIR}~g" ${cmds003} > cmds003 &&
```
Comment (SteVwonder) [inline t/t3006-resource-advanced.t]: TIL that whatever follows `s` in sed is used as the delimiter.  This is super useful!

EDIT - no need to change anything here, just wanted to remark on how neat this is

File: t/t3006-resource-advanced.t (hunk 1 above)
Comment (dongahn) [reply t/t3006-resource-advanced.t]: Good catch! Thanks.

Comment (dongahn) [reply t/t3006-resource-advanced.t]: Ok. I made this change and squash it w/ the last commit. After PR #455 is merged, I will rebase this to upstream/master and force a push.

**Comment by dongahn** [comment]
@SteVwonder: Ok. I rebased this to the upstream master and forced a push. I believe this is ready to be merged once Travis turns green. Thanks!

**Comment by SteVwonder** [comment]
Thanks @dongahn.  LGTM! This is a really cool demonstration of the flexibility and power of the scheduler.

Merging.

**Comment by dongahn** [comment]
Thanks @StevWonder! Yeah exciting time.

**Pull request merged**